libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -3409,6 +3410,7 @@ CK_DECLARE_FUNCTION( CK_RV, C_Sign )( CK_SESSION_HANDLE xSession,
     CK_BBOOL xSignatureGenerated = CK_FALSE;
     uint8_t ecSignature[ pkcs11ECDSA_P256_SIGNATURE_LENGTH + 15 ]; /*TODO: Figure out this length. */
     int lMbedTLSResult;
+    char cMbedTlsErrorStr[ 134 ];
```

Comment:
This increases the stack size quite a bit, I don't think we want to do this.

Reply:
To add to this, this seems like a magic number, I think we should use some kind of macro to define the size, were does 134 come from?

Reply:
@gkwicker perhaps we could use a global buffer, and protect it with a mutex? Or allocate a buffer on the heap in the case we do receive an error.

Reply:
Yes, good suggestion to use a macro to avoid magic numbers. I came up with 134 as size of the largest error message in [error.c](https://github.com/aws/amazon-freertos/blob/master/libraries/3rdparty/mbedtls/library/error.c#L409)(hyperlinked to the longest error string) + 1 (for the NULL character)

Reply:
I thought about using file-specific globals to avoid allocation the buffer on the stack, but wasn't sure whether that would require shared memory protection for thread-safety (in cases of multiple tls connections being alive on the platform)